specifyweb/frontend/js_src/lib/components/Merging/index.tsx

```diff
@@ -278,21 +279,49 @@ function MergeButton<SCHEMA extends AnySchema>({
     );
   }, [mergeResource]);
 
+  const [warningDialog,_,handleCloseWarningDialog, toggleWarningDialog] = useBooleanState(false)
```

Comment:
can simplify the logic and improve UX:
as soon as the user dismisses the warning, trigger the form submit - save user a click and removes the need for isAccepted state